orange/src/main/java/com/hartwig/hmftools/orange/OrangeRNAConfig.java

```diff
@@ -19,93 +28,76 @@ public interface OrangeRNAConfig {
 
     String RNA_SAMPLE_ID = "rna_sample_id";
 
-    String ISOFOX_GENE_DISTRIBUTION_CSV = "isofox_gene_distribution_csv";
-    String ISOFOX_ALT_SJ_COHORT_CSV = "isofox_alt_sj_cohort_csv";
-
-    String ISOFOX_SUMMARY_CSV = "isofox_summary_csv";
-    String ISOFOX_GENE_DATA_CSV = "isofox_gene_data_csv";
-    String ISOFOX_FUSION_CSV = "isofox_fusion_csv";
-    String ISOFOX_ALT_SPLICE_JUNCTION_CSV = "isofox_alt_splice_junction_csv";
+    String ISOFOX_GENE_DISTRIBUTION_CSV = "isofox_gene_distribution";
+    String ISOFOX_ALT_SJ_COHORT_CSV = "isofox_alt_sj_cohort";
 
     @NotNull
-    static Options createOptions() {
-        Options options = new Options();
-
-        options.addOption(RNA_SAMPLE_ID, true, "(Optional) The RNA sample of the tumor sample for which ORANGE will run.");
+    static void registerConfig(final ConfigBuilder configBuilder) {
 
-        options.addOption(ISOFOX_GENE_DISTRIBUTION_CSV, true, "(Optional) Path to isofox gene distribution CSV.");
-        options.addOption(ISOFOX_ALT_SJ_COHORT_CSV, true, "(Optional) Path to isofox alt SJ cohort CSV.");
+        configBuilder.addConfigItem(RNA_SAMPLE_ID, "(Optional) The RNA sample of the tumor sample for which ORANGE will run");
 
-        options.addOption(ISOFOX_SUMMARY_CSV, true, "(Optional) Path towards the ISOFOX summary data.");
-        options.addOption(ISOFOX_GENE_DATA_CSV, true, "(Optional) Path towards the ISOFOX gene data.");
-        options.addOption(ISOFOX_FUSION_CSV, true, "(Optional) Path towards the ISOFOX fusion data.");
-        options.addOption(ISOFOX_ALT_SPLICE_JUNCTION_CSV, true, "(Optional) Path towards the ISOFOX alt splice junction data.");
+        configBuilder.addPath(ISOFOX_GENE_DISTRIBUTION_CSV, false, "(Optional) Path to isofox gene distribution CSV");
+        configBuilder.addPath(ISOFOX_ALT_SJ_COHORT_CSV, false, "(Optional) Path to isofox alt SJ cohort CSV.");
 
-        return options;
+        configBuilder.addPath(ISOFOX_DIR_CFG, false, ISOFOX_DIR_DESC);
     }
 
-    @NotNull
     String rnaSampleId();
-
-    @NotNull
     String isofoxGeneDistributionCsv();
-
-    @NotNull
     String isofoxAltSjCohortCsv();
-
-    @NotNull
     String isofoxSummaryCsv();
-
-    @NotNull
     String isofoxGeneDataCsv();
-
-    @NotNull
     String isofoxFusionCsv();
-
-    @NotNull
     String isofoxAltSpliceJunctionCsv();
 
-    @NotNull
-    static OrangeRNAConfig createConfig(@NotNull CommandLine cmd) throws ParseException {
-        if (!hasCompleteRnaConfig(cmd,
-                RNA_SAMPLE_ID,
-                ISOFOX_GENE_DISTRIBUTION_CSV,
-                ISOFOX_ALT_SJ_COHORT_CSV,
-                ISOFOX_SUMMARY_CSV,
-                ISOFOX_GENE_DATA_CSV,
-                ISOFOX_FUSION_CSV,
-                ISOFOX_ALT_SPLICE_JUNCTION_CSV)) {
-            LOGGER.debug("No proper RNA input fed to ORANGE");
+    static OrangeRNAConfig createConfig(final ConfigBuilder configBuilder) {
```

Comment:
Guess this should be annotated @Nullable? Or is that not part of the current hmftools code standard?

Reply:
It shouldn't be @Nullable, no - it is correct as-is.